src/Core/SettingsChangesHistory.cpp

```diff
@@ -582,6 +582,7 @@ static std::initializer_list<std::pair<ClickHouseVersion, SettingsChangesHistory
 {
     {"24.12",
         {
+            {"enforce_index_structure_match_on_partition_manipulation", true, false, "Add new setting to allow attach when source table's projections and secondary indices is a subset of those in the target table."}
```

Comment:
The default for old services looks incorrect. With this it will be enabled by default for old compatibility values. Is that correct?